cpp/src/parquet/column_reader.h

```diff
@@ -151,7 +151,7 @@ class PARQUET_EXPORT PageReader {
   }
 
   // @returns: shared_ptr<Page>(nullptr) on EOS, std::shared_ptr<Page>
-  // containing new Page otherwise
+  // containing new Page otherwise. This object must not outlive the PageReader.
```

Comment:
Why, seems that `shared_ptr<Page>` can outlives PageReader, isn't it just a shared buffer?

Reply:
I would say it is fine if the buffer (as well as the page object) outlives the PageReader. But its lifecycle is bound to the memory pool which allocates the buffer.

Reply:
Sorry for the late response. This comment may be incorrect. According to https://github.com/apache/arrow/issues/14748#issuecomment-1333336429, I think [emkornfield](https://github.com/emkornfield)'s meaning is that, we need to make sure that the object "Column(i)" must outlive the "Column(i).get()", right? If so, this comment should be "shared_ptr\<Page\>" must outlive its pointer contained (Page *).
I'm not that familiar with C++. I don't know if I understand correctly or not, if not, please let me know. Thank you.